packages/corejs/src/monitoring/index.tsx

```diff
@@ -0,0 +1,130 @@
+import * as Sentry from '@sentry/browser';
+import { BrowserTracing } from '@sentry/tracing';
+import Constants from 'expo-constants';
+import React, { createContext, useCallback, useContext, useEffect, useMemo } from 'react';
+import { Primitive } from '@sentry/types';
+
+import { MonitoringConfigType, MonitoringContextType, SentryTransactionStatus } from './types';
+
+export * from './types';
+
+/**
+ * Monitoring context which will create wrapper for monitoring functionality.
+*/
+export const MonitoringContext = createContext<MonitoringContextType | null>(null);
+
+/**
+ * Monitoring wrapper used to abstract Sentry functionality.
+ *
+ * @param {object} children - Provider children components that will be shared inside context provider.
+ * @param {MonitoringConfig} config - Configuration for sentry to override default configuration.
+*/
+export function MonitoringProvider({ children, config }: MonitoringConfigType) {
+  useEffect(() => {
+    Sentry.init({
+      dsn: (config.dsn || Constants.manifest.extra.SENTRY_DSN) as string,
+      environment: (config.environment || Constants.manifest.extra.ENV) as string,
+      debug: config.debug || Constants.manifest.extra.ENV !== 'production',
+      tracesSampleRate: config.tracesSampleRate || 0.5,
+      integrations: [
+        new BrowserTracing({ tracePropagationTargets: config.tracingOrigins }),
+      ],
+    });
+  }, []);
+
+  /**
+   * Updates user context information for future events.
+   *
+   * @param id {string} set user for in sentry
+  */
+  const setMonitoringUser = useCallback((id: string): void => {
+    Sentry.setUser({ id });
+  }, []);
+
+  /**
+   * Set key:value that will be sent as tags data with the event.
+   *
+   * Can also be used to unset a tag, by passing `undefined`.
+   *
+   * @param key String key of tag
+   * @param value Value of tag
+  */
+  const setMonitoringTag = useCallback((key: string, value: Primitive): void => {
+    Sentry.setTag(key, value);
+  }, []);
+
+  /**
+   * Error handler function which is used to capture errors in sentry.
+   *
+   * @param error {Error | string} - Caught error that to be send to Sentry.io
+   */
+  const errorHandler = useCallback((error: Error | string): string | null => {
+    if (!Sentry) {
+      return null;
+    }
+
+    return Sentry.captureException(error);
+  }, []);
+
+  /**
+   * Measure the performance of application based on functionality and operation based on it.
+   * Return type of the function is the IIFE, which will helps to close the transaction and complete the measurement.
+   *
+   * @param name {string} - Name of transaction
+   * @param operation {string} - Operation of transaction to be performed
+   * @param data {{[key: string]: number | string}} - Data to be added on transaction
+   */
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  const measurePerformance = useCallback((name: string, op: string, data?: { [key: string]: number | string }): Function => {
```

Comment:
The @typescript-eslint/ban-types rule is there for a reason ! The `Function` type should NOT be used. Please remove the eslint comment and replace the `Function` type with a type definition for the callback returned